cedar-policy/src/api.rs

```diff
@@ -1385,35 +1387,143 @@ impl<'a> ValidationResult<'a> {
     }
 
     /// Get an iterator over the errors found by the validator.
-    pub fn validation_errors(&self) -> impl Iterator<Item = &ValidationError<'a>> {
+    pub fn validation_errors(&self) -> impl Iterator<Item = &ValidationError<'static>> {
         self.validation_errors.iter()
     }
 
     /// Get an iterator over the warnings found by the validator.
-    pub fn validation_warnings(&self) -> impl Iterator<Item = &ValidationWarning<'a>> {
+    pub fn validation_warnings(&self) -> impl Iterator<Item = &ValidationWarning<'static>> {
         self.validation_warnings.iter()
     }
+
+    fn first_error_or_warning(
+        &self,
+    ) -> Option<Either<&ValidationError<'a>, &ValidationWarning<'a>>> {
```

Comment:
would it simplify things to return `Option<&dyn Diagnostic>` instead of `Option<Either<...>>`?  do any callers need to discriminate between error vs warning?

Reply:
I think that will be easier